tools/cldr-code/src/main/java/org/unicode/cldr/util/CalculatedCoverageLevels.java

```diff
@@ -26,6 +31,30 @@ public Map<String, Level> getLevels() {
         return levels;
     }
 
+    public Level getEffectiveCoverageLevel(String locale) {
+        return getEffectiveCoverageLevel(CLDRLocale.getInstance(locale));
+    }
+
+    public Level getEffectiveCoverageLevel(CLDRLocale locale) {
+        // per spec, assumed level for the explicit root locale
+        if (locale == CLDRLocale.ROOT) {
+            return DEFAULT_ROOT_LEVEL;
+        }
+        // See if there is an explicit entry
+        final Level level = levels.get(locale.getBaseName());
```

Comment:
It should follow the locale idparser getparent chain

Reply:
It does via CLDRLocale